diff --git a/server/dialect-daco/src/main/antlr4/org/eclipse/lsp/cobol/dialects/daco/TechnicalLexer.g4 b/server/dialect-daco/src/main/antlr4/org/eclipse/lsp/cobol/dialects/daco/TechnicalLexer.g4
index 65addce..5051808 100644
--- a/server/dialect-daco/src/main/antlr4/org/eclipse/lsp/cobol/dialects/daco/TechnicalLexer.g4
+++ b/server/dialect-daco/src/main/antlr4/org/eclipse/lsp/cobol/dialects/daco/TechnicalLexer.g4
@@ -69,8 +69,8 @@ NONNUMERICLITERAL : UNTRMSTRINGLITERAL | STRINGLITERAL | DBCSLITERAL | HEXNUMBER
 
 CHAR_STRING_CONSTANT : HEXNUMBER | STRINGLITERAL;
 
-IDENTIFIER : [a-zA-Z0-9][-_a-zA-Z0-9]*;
-COPYBOOK_IDENTIFIER : [a-zA-Z0-9#@$][-_a-zA-Z0-9#@$]*;
+IDENTIFIER : [a-zA-Z0-9\u3040-\u309F\u30A0-\u30FF\u4E00-\u9FFF\uFF10-\uFF19\uFF21-\uFF3A\uFF41-\uFF5A\uFF66-\uFF9F][-_a-zA-Z0-9\u3040-\u309F\u30A0-\u30FF\u4E00-\u9FFF\uFF10-\uFF19\uFF21-\uFF3A\uFF41-\uFF5A\uFF66-\uFF9F]*;
+COPYBOOK_IDENTIFIER : [a-zA-Z0-9#@$\u3040-\u309F\u30A0-\u30FF\u4E00-\u9FFF\uFF10-\uFF19\uFF21-\uFF3A\uFF41-\uFF5A\uFF66-\uFF9F][-_a-zA-Z0-9#@$\u3040-\u309F\u30A0-\u30FF\u4E00-\u9FFF\uFF10-\uFF19\uFF21-\uFF3A\uFF41-\uFF5A\uFF66-\uFF9F]*;
 FILENAME : IDENTIFIER+ '.' IDENTIFIER+;
 
 OCTDIGITS : OCT_DIGIT;
diff --git a/server/dialect-idms/src/main/antlr4/org/eclipse/lsp/cobol/dialects/idms/IdmsTechnicalLexer.g4 b/server/dialect-idms/src/main/antlr4/org/eclipse/lsp/cobol/dialects/idms/IdmsTechnicalLexer.g4
index c1163d9..d996025 100644
--- a/server/dialect-idms/src/main/antlr4/org/eclipse/lsp/cobol/dialects/idms/IdmsTechnicalLexer.g4
+++ b/server/dialect-idms/src/main/antlr4/org/eclipse/lsp/cobol/dialects/idms/IdmsTechnicalLexer.g4
@@ -69,8 +69,8 @@ NONNUMERICLITERAL : UNTRMSTRINGLITERAL | STRINGLITERAL | DBCSLITERAL | HEXNUMBER
 
 CHAR_STRING_CONSTANT : HEXNUMBER | STRINGLITERAL;
 
-IDENTIFIER : [a-zA-Z0-9][-_a-zA-Z0-9]*;
-COPYBOOK_IDENTIFIER : [a-zA-Z0-9#@$][-_a-zA-Z0-9#@$]*;
+IDENTIFIER : [a-zA-Z0-9\u3040-\u309F\u30A0-\u30FF\u4E00-\u9FFF\uFF10-\uFF19\uFF21-\uFF3A\uFF41-\uFF5A\uFF66-\uFF9F][-_a-zA-Z0-9\u3040-\u309F\u30A0-\u30FF\u4E00-\u9FFF\uFF10-\uFF19\uFF21-\uFF3A\uFF41-\uFF5A\uFF66-\uFF9F]*;
+COPYBOOK_IDENTIFIER : [a-zA-Z0-9#@$\u3040-\u309F\u30A0-\u30FF\u4E00-\u9FFF\uFF10-\uFF19\uFF21-\uFF3A\uFF41-\uFF5A\uFF66-\uFF9F][-_a-zA-Z0-9#@$\u3040-\u309F\u30A0-\u30FF\u4E00-\u9FFF\uFF10-\uFF19\uFF21-\uFF3A\uFF41-\uFF5A\uFF66-\uFF9F]*;
 FILENAME : IDENTIFIER+ '.' IDENTIFIER+;
 
 OCTDIGITS : OCT_DIGIT;
diff --git a/server/engine/src/main/antlr4/org/eclipse/lsp/cobol/core/CompilerDirectivesLexer.g4 b/server/engine/src/main/antlr4/org/eclipse/lsp/cobol/core/CompilerDirectivesLexer.g4
index 1604404..58ca4ab 100644
--- a/server/engine/src/main/antlr4/org/eclipse/lsp/cobol/core/CompilerDirectivesLexer.g4
+++ b/server/engine/src/main/antlr4/org/eclipse/lsp/cobol/core/CompilerDirectivesLexer.g4
@@ -436,7 +436,7 @@ LPARENCHAR : '(';
 RPARENCHAR : ')';
 DOT: '.';
 
-IDENTIFIER : [a-zA-Z0-9][-_a-zA-Z0-9]*;
+IDENTIFIER : [a-zA-Z0-9\u3040-\u309F\u30A0-\u30FF\u4E00-\u9FFF\uFF10-\uFF19\uFF21-\uFF3A\uFF41-\uFF5A\uFF66-\uFF9F][-_a-zA-Z0-9\u3040-\u309F\u30A0-\u30FF\u4E00-\u9FFF\uFF10-\uFF19\uFF21-\uFF3A\uFF41-\uFF5A\uFF66-\uFF9F]*;
 FILENAME : IDENTIFIER+ '.' IDENTIFIER+;
 
 fragment STRINGLITERAL:
diff --git a/server/engine/src/main/antlr4/org/eclipse/lsp/cobol/core/TechnicalLexer.g4 b/server/engine/src/main/antlr4/org/eclipse/lsp/cobol/core/TechnicalLexer.g4
index 3004125..f9777a0 100644
--- a/server/engine/src/main/antlr4/org/eclipse/lsp/cobol/core/TechnicalLexer.g4
+++ b/server/engine/src/main/antlr4/org/eclipse/lsp/cobol/core/TechnicalLexer.g4
@@ -69,8 +69,8 @@ NONNUMERICLITERAL : UNTRMSTRINGLITERAL | STRINGLITERAL | DBCSLITERAL | HEXNUMBER
 
 CHAR_STRING_CONSTANT : HEXNUMBER | STRINGLITERAL;
 
-IDENTIFIER : [a-zA-Z0-9][-_a-zA-Z0-9]*;
-COPYBOOK_IDENTIFIER : [a-zA-Z0-9#@$][-_a-zA-Z0-9#@$]*;
+IDENTIFIER : [a-zA-Z0-9\u3040-\u309F\u30A0-\u30FF\u4E00-\u9FFF\uFF10-\uFF19\uFF21-\uFF3A\uFF41-\uFF5A\uFF66-\uFF9F][-_a-zA-Z0-9\u3040-\u309F\u30A0-\u30FF\u4E00-\u9FFF\uFF10-\uFF19\uFF21-\uFF3A\uFF41-\uFF5A\uFF66-\uFF9F]*;
+COPYBOOK_IDENTIFIER : [a-zA-Z0-9#@$\u3040-\u309F\u30A0-\u30FF\u4E00-\u9FFF\uFF10-\uFF19\uFF21-\uFF3A\uFF41-\uFF5A\uFF66-\uFF9F][-_a-zA-Z0-9#@$\u3040-\u309F\u30A0-\u30FF\u4E00-\u9FFF\uFF10-\uFF19\uFF21-\uFF3A\uFF41-\uFF5A\uFF66-\uFF9F]*;
 FILENAME : IDENTIFIER+ '.' IDENTIFIER+;
 
 OCTDIGITS : OCT_DIGIT;
diff --git a/server/engine/src/main/antlr4/org/eclipse/lsp/cobol/implicitDialects/cics/CICSLexer.g4 b/server/engine/src/main/antlr4/org/eclipse/lsp/cobol/implicitDialects/cics/CICSLexer.g4
index 3620da8..5d1728d 100644
--- a/server/engine/src/main/antlr4/org/eclipse/lsp/cobol/implicitDialects/cics/CICSLexer.g4
+++ b/server/engine/src/main/antlr4/org/eclipse/lsp/cobol/implicitDialects/cics/CICSLexer.g4
@@ -2895,8 +2895,8 @@ NUMERICLITERAL : (PLUSCHAR | MINUSCHAR)? DIGIT* (DOT | COMMACHAR) DIGIT+ (('e' |
 
 NONNUMERICLITERAL : UNTRMSTRINGLITERAL | STRINGLITERAL | DBCSLITERAL | HEXNUMBER | NULLTERMINATED;
 
-WORD_IDENTIFIER : [a-zA-Z0-9][-_a-zA-Z0-9]*;
-COPYBOOK_IDENTIFIER : [a-zA-Z0-9#@$][-_a-zA-Z0-9#@$]*;
+WORD_IDENTIFIER : [a-zA-Z0-9\u3040-\u309F\u30A0-\u30FF\u4E00-\u9FFF\uFF10-\uFF19\uFF21-\uFF3A\uFF41-\uFF5A\uFF66-\uFF9F][-_a-zA-Z0-9\u3040-\u309F\u30A0-\u30FF\u4E00-\u9FFF\uFF10-\uFF19\uFF21-\uFF3A\uFF41-\uFF5A\uFF66-\uFF9F]*;
+COPYBOOK_IDENTIFIER : [a-zA-Z0-9#@$\u3040-\u309F\u30A0-\u30FF\u4E00-\u9FFF\uFF10-\uFF19\uFF21-\uFF3A\uFF41-\uFF5A\uFF66-\uFF9F][-_a-zA-Z0-9#@$\u3040-\u309F\u30A0-\u30FF\u4E00-\u9FFF\uFF10-\uFF19\uFF21-\uFF3A\uFF41-\uFF5A\uFF66-\uFF9F]*;
 
 OCTDIGITS : OCT_DIGIT;
 HEX_NUMBERS : HEXNUMBER;
diff --git a/server/engine/src/main/java/org/eclipse/lsp/cobol/core/preprocessor/delegates/replacement/ReplacingServiceImpl.java b/server/engine/src/main/java/org/eclipse/lsp/cobol/core/preprocessor/delegates/replacement/ReplacingServiceImpl.java
index ccef276..924f87b 100644
--- a/server/engine/src/main/java/org/eclipse/lsp/cobol/core/preprocessor/delegates/replacement/ReplacingServiceImpl.java
+++ b/server/engine/src/main/java/org/eclipse/lsp/cobol/core/preprocessor/delegates/replacement/ReplacingServiceImpl.java
@@ -170,7 +170,10 @@ public class ReplacingServiceImpl implements ReplacingService {
     String text = extendedDocument.getBaseText(startLine, endLine);
     final List<Integer> lineMap = makeLineMap(text);
     try {
-      Matcher matcher = Pattern.compile(pattern.getLeft(), Pattern.CASE_INSENSITIVE).matcher(text);
+      Matcher matcher =
+          Pattern.compile(
+                  pattern.getLeft(), Pattern.CASE_INSENSITIVE | Pattern.UNICODE_CHARACTER_CLASS)
+              .matcher(text);
       while (matcher.find()) {
         Range range =
             new Range(
diff --git a/server/parser/src/main/antlr4/org/eclipse/lsp/cobol/core/CobolLexer.g4 b/server/parser/src/main/antlr4/org/eclipse/lsp/cobol/core/CobolLexer.g4
index 5a9427b..b59987a 100644
--- a/server/parser/src/main/antlr4/org/eclipse/lsp/cobol/core/CobolLexer.g4
+++ b/server/parser/src/main/antlr4/org/eclipse/lsp/cobol/core/CobolLexer.g4
@@ -897,7 +897,7 @@ UNKNOWN_EXEC : E X E C [ \n\r]+
         )
         ;
 
-IDENTIFIER : [a-zA-Z0-9][-_a-zA-Z0-9]*;
+IDENTIFIER : [a-zA-Z0-9\u3040-\u309F\u30A0-\u30FF\u4E00-\u9FFF\uFF10-\uFF19\uFF21-\uFF3A\uFF41-\uFF5A\uFF66-\uFF9F][-_a-zA-Z0-9\u3040-\u309F\u30A0-\u30FF\u4E00-\u9FFF\uFF10-\uFF19\uFF21-\uFF3A\uFF41-\uFF5A\uFF66-\uFF9F]*;
 
 // treat all the non-processed tokens as errors
 ERRORCHAR : . ;
